src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -343,6 +356,53 @@ private void validateOperationProperties(@Nullable Schema schema, FailureCollect
           .withConfigElement(NAME_TABLE_KEY, keyField);
       }
     }
+
+    Map<String, Integer> keyMap = calculateDuplicates(keyFields);
+    keyMap.keySet().stream()
+      .filter(key -> keyMap.get(key) != 1)
+      .forEach(key -> collector.addFailure(
+        String.format("Table key field '%s' is duplicated.", key),
+        String.format("Remove duplicates of Table key field '%s'.", key))
+        .withConfigElement(NAME_TABLE_KEY, key)
+      );
+
+    if (Operation.UPSERT.equals(getOperation()) && getOrderedBy() != null) {
+      List<String> orderedByList = Arrays.stream(Objects.requireNonNull(getOrderedBy()).split(","))
+        .map(String::trim).collect(Collectors.toList());
+      LOG.debug("-> " + orderedByList);
+      orderedByList.stream()
+        .filter(v -> !fields.contains(v.split(":")[0]))
+        .forEach(v -> collector.addFailure(
+        String.format("Ordered by field '%s' does not exist in the schema.", v.split(":")[0]),
+        "Change the Ordered by field to be one of the schema fields.")
+        .withConfigElement(NAME_ORDERED_BY, v));
+
+      Map<String, Integer> orderedByFieldMap = calculateDuplicates(orderedByList);
+      Map<String, String> orderedByFieldValueMap = orderedByList.stream()
+        .collect(Collectors.toMap(p -> p.split(":")[0], p -> p, (x, y) -> y));
+
+      orderedByFieldMap.keySet().stream()
+        .filter(key -> orderedByFieldMap.get(key) != 1)
+        .forEach(key -> collector.addFailure(
+          String.format("Ordered by field '%s' is duplicated.", key),
+          String.format("Remove duplicates of Ordered by field '%s'.", key))
+          .withConfigElement(NAME_ORDERED_BY, orderedByFieldValueMap.get(key))
+        );
+
+      orderedByList.stream()
+        .filter(v -> v.split(":").length != 2)
+        .forEach(v -> collector.addFailure(
+          "Missing sort order.",
```

Comment:
this seems like it should just be defaulted (to whatever order is default in bigquery). We wouldn't expect this when the pipeline is created from the UI, but it could happen when created through the REST API.

Reply:
This validation was removed.